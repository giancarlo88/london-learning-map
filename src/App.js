import React, { Component } from 'react';
import { GoogleMap, GoogleMapLoader, Marker } from "react-google-maps";
import { default as _ } from "lodash";
import './App.css';
import Info from './info.js';
import CMS from './cms.js';
import $ from 'jquery';


class Map extends Component {
  state = {
    markers: [], 
    cms: false
  }
  handleMarkerClick (targetMarker) {
  this.setState({
    markers: this.state.markers.map(marker => {
      if (marker === targetMarker) {
        return {
          ...marker,
          showInfo: true,
          opacity:.5,
        };
      } else {
        return {
          ...marker,
          showInfo: false,
          opacity: 0.5
        }
      }
    }),
  });
  }
  handleMarkerClose (targetMarker) {
    this.setState({
      markers: this.state.markers.map(marker => {
        if (marker === targetMarker) {
          return {
            ...marker,
            showInfo: false,
          };
        }
        return marker;
      }),
    });
  }
  toggleCMS = () => {
    this.setState({ 
      cms: !this.state.cms
    })
  }
  componentDidMount(){
    $.ajax({
        url: "http://www.ggalliani.com/projects/llm/api.php/map?transform=1",
        method: "GET",
        dataType: 'json',
        success: function(data) {
          this.setState({markers: data.map})
          }.bind(this),
        error: function(xhr, status, err) {
          console.error(this.props.url, status, err.toString());
          }.bind(this)
        })
  }
  renderInfoWindow(ref, marker) {
    const onCloseclick = this.handleMarkerClose.bind(this, marker);
    console.log(ref)
    return (
      <Info
        title={this.state.markers[ref].title}
        info={this.state.markers[ref].info}
        key={`${ref}_info_window`}
        onCloseclick={onCloseclick}
      >
      {this.state.markers[ref].info}
      </Info>
      )
  }
  render() {
    const {markers} = this.state
     return (
      <div>
      {!this.state.cms 
        && <GoogleMapLoader
              query={{ libraries: "geometry,drawing,places,visualization" }}
              containerElement={
                <div 
                  {...this.props}
                  style={{
                    height: `100vh`,
                    width: `100 vw`
                  }}
                />
              }
              googleMapElement={
                <GoogleMap
                  ref={(map) => console.log(map)}
                  defaultZoom={12}
                  defaultCenter={{ lat: 51.511507, lng:-0.115705}}
                >
                  {markers.map((marker, index) => {
                    const ref = index;
                    const onClick = () => this.handleMarkerClick(marker);
                    return (
                    
                      <Marker
                        icon={marker.showInfo ? '/assets/blue-pushpin.png' : null}
                        title={marker.title}
                        key={index}
                        position={ 
                          { lat: Number(marker['x-cord']), 
                            lng: Number(marker['y-cord'])
                          }
                        }
                        onClick={onClick}
                      >
                        {marker.showInfo ? this.renderInfoWindow(ref, marker) : null}
                        </Marker>
                      ) 
                    }
                  )}
                <button className='toggle-cms' onClick={this.toggleCMS}>Add a location...</button>
                </GoogleMap>
              }
            />
        }
      {this.state.cms && <CMS toggleCMS={this.toggleCMS} markers={this.state.markers}/> }
    </div>
    );
  }
}

export default Map;
