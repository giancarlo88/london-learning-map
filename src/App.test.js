import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

describe(App.name, () => {
  describe('@renders', () => {
    it('should match a snapshot', () => {
      const wrapper = shallow(<App />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
