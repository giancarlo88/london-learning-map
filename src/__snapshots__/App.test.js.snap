// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App @renders should match a snapshot 1`] = `<MapContainer />`;
